java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -478,24 +477,11 @@ private Resolution findMethod(Env env, JavaType callSite, JavaType site, String
     return bestSoFar;
   }
 
-  private JavaType handleTypeArguments(List<JavaType> typeParams, JavaType type, JavaSymbol.MethodJavaSymbol methodSymbol) {
-    if (!typeParams.isEmpty() && methodSymbol.typeVariableTypes.size() == typeParams.size()) {
-      TypeSubstitution typeSubstitution = new TypeSubstitution();
-      int i = 0;
-      for (JavaType.TypeVariableJavaType typeVariableType : methodSymbol.typeVariableTypes) {
-        typeSubstitution.add(typeVariableType, typeParams.get(i));
-        i++;
-      }
-      return substituteTypeParameter(type, typeSubstitution);
-    }
-    return type;
-  }
-
   /**
    * @param candidate    candidate
    * @param bestSoFar previously found best match
    */
-  private JavaSymbol selectBest(Env env, JavaType site, List<JavaType> argTypes, JavaSymbol candidate, JavaSymbol bestSoFar, boolean autoboxing) {
+  private JavaSymbol selectBest(Env env, JavaType site, List<JavaType> argTypes, List<JavaType> typeParams, JavaSymbol candidate, JavaSymbol bestSoFar, boolean autoboxing) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "selectBest" is 16 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)